hack/generator/pkg/codegen/pipeline/flatten_properties.go

```diff
@@ -89,17 +87,42 @@ func removeFlatten(t astmodel.Type) astmodel.Type {
 	return t
 }
 
-func checkForDuplicateNames(props []*astmodel.PropertyDefinition) error {
-	names := make(map[astmodel.PropertyName]struct{})
+func fixCollisions(props []*astmodel.PropertyDefinition) []*astmodel.PropertyDefinition {
```

Comment:
minor: Maybe throw a comment on here with the example you have in the PR description? `Type` + `Properties` + `Type`.